core/src/plugins/filed/python/postgresql/bareos-fd-postgresql.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# BAREOS - Backup Archiving REcovery Open Sourced
+#
+# Copyright (C) 2023-2023 Bareos GmbH & Co. KG
+#
+# This program is Free Software; you can redistribute it and/or
+# modify it under the terms of version three of the GNU Affero General Public
+# License as published by the Free Software Foundation, which is
+# listed in the file LICENSE.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.
+#
+# Author: Bareos Team
+#
+"""bareos-fd-postgresql is a plugin to backup PostgreSQL cluster
+using BareosFdPluginPostgreSQL
+"""
+from sys import version_info
+
+# Provided by the Bareos FD Python plugin interface
+import bareosfd
+from bareosfd import *
+
+# This module contains the wrapper functions called by the Bareos-FD, the
+# functions call the corresponding methods from your plugin class
+import BareosFdWrapper
+
+# from BareosFdWrapper import parse_plugin_definition, handle_plugin_event,
+# start_backup_file, end_backup_file, start_restore_file, end_restore_file,
+# restore_object_data, plugin_io, create_file, check_file, handle_backup_file
+# # noqa
+
+from BareosFdWrapper import *  # noqa
+
+
+def load_bareos_plugin(plugindef):
```

Comment:
we have a default for `load_bareos_plugin()` that can be used via a class decorator.
Using that you can easily have plugins in one single file only, removing the need to maintain two files.
All additional initialization code can be added directly to the module so it runs immediately on module load.

If you want, we can discuss this or I can prepare a commit that applies the changes.
I guess we also wanted to change all the existing plugins from the two-file-approach to one single file.

Reply:
I'm in favor of doing one module, so please yes, either we discuss of that, or you propose a patch for.